src/main/java/ttt/CommandPrompt.java

```diff
@@ -67,44 +74,141 @@ public void print(Board board) {
 
     @Override
     public void printWinningMessageFor(PlayerSymbol symbol) {
-        display("Congratulations - " + symbol + " has won" + newLine());
+        display(FONT_COLOUR_ANSII_CHARACTERS
+                + "Congratulations - "
+                + colour(symbol)
+                + FONT_COLOUR_ANSII_CHARACTERS
+                + " has won");
     }
 
     @Override
     public void printDrawMessage() {
-        display("No winner this time" + newLine());
+        display(FONT_COLOUR_ANSII_CHARACTERS
+                + "No winner this time");
     }
 
-    @Override
-    public void clear() {
-        display(CLEAR_SCREEN_ANSII_CHARACTERS);
+    private Function<ValidationResult, Void> functionToRepromptGameType() {
+        return validationResult -> {
+            clear();
+            display(BOARD_COLOUR_ANSII_CHARACTERS + validationResult.reason());
+            askUserForGameType();
+            return null;
+        };
+    }
+
+    private Function<ValidationResult, Void> functionToRepromptReplay() {
+        return validationResult -> {
+            clear();
+            display(BOARD_COLOUR_ANSII_CHARACTERS + validationResult.reason());
+            askUserToPlayAgain();
+            return null;
+        };
+    }
+
+    private int asInteger(String input) {
+        return Integer.valueOf(input);
+    }
+
+    private List<InputValidator> gameTypeValidators() {
+        return Arrays.asList(new NumericValidator(), new GameTypeValidator());
+    }
+
+    private String getValidInput(InputValidator compoundValidator, String input, Function reprompt) {
+        ValidationResult validationResult = compoundValidator.isValid(input);
+        while (!validationResult.isValid()) {
+            reprompt.apply(validationResult);
+            validationResult = compoundValidator.isValid(input());
+        }
+        clear();
+        return validationResult.userInput();
+    }
+
+    private String getValidMove(InputValidator compoundValidator,
```

Comment:
This loop is almost identical to the one above, but because of the Board object being used, I am unable to generalise the two of them. The Function would need to take a Board and a ValidationResult which is not possible.